Sources/core/objects/Gun.cpp

```diff
@@ -0,0 +1,67 @@
+#include "Gun.h"
+
+Gun::Gun(size_t id) : Item(id)
+{
+    SetSprite("icons/guns.dmi");
+
+}
+
+bool Gun::Handle_Ammo()
+{
+    if (ammunition_ <= 0)
+    {
+        ammunition_ = 0;
+        return false;
+    }
+    --ammunition_;
+    return true;
+}
+
+void Gun::Shoot(VDir target,id_ptr_on<Human> shooter)
+{	
```

Comment:
Not sure that `shooter` variable is needed. As far as I understand, `GetOwner()` is always equal to `shooter`.